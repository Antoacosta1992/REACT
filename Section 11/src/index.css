body {
    padding: 20px;
    color:#66CDAA;
}
button {
    margin-right: 10px;

}

a {
    margin-right: 5px;
}