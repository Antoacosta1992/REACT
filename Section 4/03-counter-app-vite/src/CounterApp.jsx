import { useState } from 'react';
import PropTypes from 'prop-types';

export const CounterApp =({value}) => {

    const [counter, setCounter] = useState ( value );

    const handleAdd = () => {
        //console.log(event)
        setCounter(counter +1);
        //setCounter ( (c) => c + 1) segunda manera.
    }
    const handleAdd1 = () =>{
        setCounter(counter-1);
    }

    const handleAdd2 = () =>{
        setCounter(value);
    }

    return (
       <>
            <h1>CounterApp</h1>
            <h2> { counter } </h2>

            <button onClick={handleAdd}> +1 </button>
            <button onClick={handleAdd1}> -1 </button>
            <button onClick={handleAdd2}> Reset </button>

       </>
    ); 
}
//cuando una funcion recibe un evento y ese evento se lo pasa directamente a otra
//funcion podemos obviar mandar la referencia, borrar esa funcion, 
//inclusive borrar el evento y solo mandamos la referencia a esa funcion.
// <button onClick={ (event ) => handleAdd(event)}> 
// <button onClick={handleAdd}>
//TAREA VIDEO 43
CounterApp.propTypes= {
    Value: PropTypes.number,}

