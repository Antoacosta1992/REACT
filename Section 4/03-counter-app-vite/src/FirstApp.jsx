import PropTypes from 'prop-types';

//import { Fragment } from 'react';

//Si fuera una funcion async saltaría error
//porque una promosa es un objeto, y ese objeto es lo que esta serielizando
// o imprimiendo en <h1>{ getResult()}</h1>
//const getResult = (a,b) => {
//    return a + b;
//}

export const FirstApp = ({
  title, 
  subTitle= 'La vida es bella',
  name
}) => {
  ////props es info que fluye desde el componente padre hacia
  //*console.log(props);
  /////este componente hijo.

//si el titulo no existe, entonces throw "error"

  return (
    <>
    <h1>{title}</h1>
    {/* <code>{JSON.stringify (newMessage)}</code> */}
    <p>{subTitle}</p>
    <p>{name}</p>
    </>
  )

  }

  FirstApp.propTypes= {
    title: PropTypes.string.isRequired,
    subTitle: PropTypes.string,
  }

FirstApp.defaultProps ={
  title: 'No hay ningún titulo',
  subTitle: 'No hay subtitulo',
  name: 'Antonela Acosta',
}