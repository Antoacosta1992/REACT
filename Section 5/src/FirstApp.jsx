import PropTypes from 'prop-types';
import React from "react";

//import { Fragment } from 'react';

//Si fuera una funcion async saltaría error
//porque una promosa es un objeto, y ese objeto es lo que esta serielizando
// o imprimiendo en <h1>{ getResult()}</h1>
//const getResult = (a,b) => {
//    return a + b;
//}

export const FirstApp = ({ title, subTitle,name }) => {
  ////props es info que fluye desde el componente padre hacia
  //*console.log(props);
  /////este componente hijo.

//si el titulo no existe, entonces throw "error"

  return (
    <>
    <h1 data-testid="test-title">{title}</h1>
    {/* <code>{JSON.stringify (newMessage)}</code> */}
    <p>{subTitle}</p>
    <p>{subTitle}</p>
    <p>{name}</p>
    </>
  )

  }

  FirstApp.propTypes= {
    title: PropTypes.string.isRequired,
    subTitle: PropTypes.string,
  }

FirstApp.defaultProps ={
  title: 'No hay ningún titulo',
  subTitle: 'No hay subtitulo',
  name: 'Antonela Acosta',
}