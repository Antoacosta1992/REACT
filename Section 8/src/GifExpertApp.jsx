import { useState } from "react";
import { AddCategory,GifGrid } from './components';

export const GifExpertApp = ()=>{

    const [categories, setCategories] = useState (['One Punch'])
    
    const onAddCategory = (newCategory) => {
        //categories.push(newCategory);
        if ( categories.includes(newCategory) ) return;
        setCategories([ newCategory, ...categories ]);
        //setCategories(cat=>[...cat, 'Valorant'])
    
    }


    return(
        <> 
        {/*TITULO*/}
        <h1>- GIF EXPERT APP - </h1>

        {/* IMPUT*/} 
        <AddCategory 
        //setCategories={setCategories}
        onNewCategory={ (value) => onAddCategory(value) }
        />

        {/*LISTADO*/}
        {/*el map me permite barrer los elemnentos del arreglo*/}
     
        {categories.map( (category) => (
                <GifGrid
                    key={category}
                    category={ category}/>

                
            ))
        }
        </>
        
    )
}