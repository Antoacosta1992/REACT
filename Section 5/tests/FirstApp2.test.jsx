import { render, screen } from '@testing-library/react';
import { FirstApp } from '../src/FirstApp';
import React from "react";

//screen = limpiezas automaticas
describe('Pruebas en <FirstApp />', () => {
    const title = 'Hola, soy Anto';
    const subTitle= 'Soy, lo que siento'

    test('Debe hacer match con Snapshot', () => { 
    
        const {container} = render( <FirstApp title={title}/>);
        expect (container).toMatchSnapshot();
     });


     test('Debe mostrar el mensaje "Hola, soy Anto"', () => { 
        render( <FirstApp title={title}/>);
        expect( screen.getByText(title)).toBeTruthy();
        //screen.debug();   para ver el objeto
      });

      test('debe de mostrar el titulo en un h1', () => {
        render( <FirstApp title={ title } /> );
        expect( screen.getByRole('heading', { level: 1 }).innerHTML ).toContain( title );

      });

      test('debe de mostrar el subtitulo enviado por props', () => {
        
        render( 
            <FirstApp 
                title={ title }
                subTitle={ subTitle }
            />  
        );

        expect( screen.getAllByText(subTitle).length ).toBe(2);

    });





    });
