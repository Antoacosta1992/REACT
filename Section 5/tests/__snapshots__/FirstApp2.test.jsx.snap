// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp /> Debe hacer match con Snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hola, soy Anto
  </h1>
  <p>
    No hay subtitulo
  </p>
  <p>
    No hay subtitulo
  </p>
  <p>
    Antonela Acosta
  </p>
</div>
`;
